### ISSUE TITLE
No motion when executing trajectories (SDA10F)

### ISSUE THREAD

**Issue by inbarbd** [ros-industrial/motoman#343]
hello, 
i work with sda10f robot with  fs100 controller with the motoman_sda10f_moveit_config. 
ros melodic
ubuntu 18.04
the MotoRos has been install and setup by Yaskawa support
software version FS100 - FS3.27.00A(US/FR)-00

when run:

```
moveit_planning_execution.launch robot_ip:=10.0.0.2 controller:=fs100
```

the robot is connected and we can read information from the controller. when move the robot manually with the pendant the robot position is updating in rviz.

when i try to plan and executed , moveit plan the path but the robot is not response. there is  NO ERRORS  

the only changes we made in the pkg is:
1. `ExecuteTrajectoryAction` relapse `MoveGroupExecuteService` in `move_group.launch`
2.  ADD `<param name="trajectory_execution/execution_duration_monitoring" value="false" />` to `jrajectory_exexution.launch.xml`
3. change topic list in `motoman_sda10f_support/sda10f_motion_interface.yaml` reon `1,2,3,4` to `0,1,2,3`

this is what is see in the terminal:

<details>

<summary>console text</summary>

```
auto-starting new master
process[master]: started with pid [23968]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to e2072170-c66d-11ea-8b5a-000acd39c600
process[rosout-1]: started with pid [23983]
started core service [/rosout]
process[joint_state-2]: started with pid [23990]
process[motion_streaming_interface-3]: started with pid [23991]
process[joint_trajectory_action-4]: started with pid [23992]
process[robot_state_publisher-5]: started with pid [23998]
process[move_group-6]: started with pid [24003]
process[rviz_roblab4_ThinkCentre_M920s_23952_6060666620694100860-7]: started with pid [24009]
[ WARN] [1594798584.454758041]: Tried to connect when socket already in connected state
[ INFO] [1594798584.459331474]: Loading robot model 'motoman_sda10f'...
[ INFO] [1594798584.493516542]: rviz version 1.13.13
[ INFO] [1594798584.493608599]: compiled against Qt version 5.9.5
[ INFO] [1594798584.493632859]: compiled against OGRE version 1.9.0 (Ghadamon)
[ INFO] [1594798584.499919164]: Forcing OpenGl version 0.
[ WARN] [1594798584.500656873]: Kinematics solver doesn't support #attempts anymore, but only a timeout.
Please remove the parameter '/robot_description_kinematics/arm_left/kinematics_solver_attempts' from your configuration.
[ERROR] [1594798584.520511248]: Group 'torso' is not a chain
[ERROR] [1594798584.520558385]: Kinematics solver of type 'kdl_kinematics_plugin/KDLKinematicsPlugin' could not be initialized for group 'torso'
[ERROR] [1594798584.520586068]: Kinematics solver could not be instantiated for joint group torso.
[ERROR] [1594798584.520599824]: Group 'arms' is not a chain
[ERROR] [1594798584.520606432]: Kinematics solver of type 'kdl_kinematics_plugin/KDLKinematicsPlugin' could not be initialized for group 'arms'
[ERROR] [1594798584.520613091]: Kinematics solver could not be instantiated for joint group arms.
[ERROR] [1594798584.520626996]: Group 'sda10f' is not a chain
[ERROR] [1594798584.520634626]: Kinematics solver of type 'kdl_kinematics_plugin/KDLKinematicsPlugin' could not be initialized for group 'sda10f'
[ERROR] [1594798584.520657186]: Kinematics solver could not be instantiated for joint group sda10f.
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
[ INFO] [1594798584.566252600]: Publishing maintained planning scene on 'monitored_planning_scene'
[ INFO] [1594798584.567378680]: MoveGroup debug mode is OFF
Starting planning scene monitors...
[ INFO] [1594798584.567395978]: Starting planning scene monitor
[ INFO] [1594798584.568290753]: Listening to '/planning_scene'
[ INFO] [1594798584.568303172]: Starting world geometry update monitor for collision objects, attached objects, octomap updates.
[ INFO] [1594798584.569117994]: Listening to '/collision_object'
[ INFO] [1594798584.569965128]: Listening to '/planning_scene_world' for planning scene world geometry
[ WARN] [1594798584.570167685]: Resolution not specified for Octomap. Assuming resolution = 0.1 instead
[ERROR] [1594798584.570325654]: Failed to find 3D sensor plugin parameters for octomap generation
[ INFO] [1594798584.576917017]: Listening to '/attached_collision_object' for attached collision objects
Planning scene monitors started.
[ INFO] [1594798584.591387117]: Initializing OMPL interface using ROS parameters
[ INFO] [1594798584.610641493]: Using planning interface 'OMPL'
[ INFO] [1594798584.612341896]: Param 'default_workspace_bounds' was not set. Using default value: 10
[ INFO] [1594798584.612509731]: Param 'start_state_max_bounds_error' was set to 0.1
[ INFO] [1594798584.612657258]: Param 'start_state_max_dt' was not set. Using default value: 0.5
[ INFO] [1594798584.612814970]: Param 'start_state_max_dt' was not set. Using default value: 0.5
[ INFO] [1594798584.612994981]: Param 'jiggle_fraction' was set to 0.05
[ INFO] [1594798584.613152770]: Param 'max_sampling_attempts' was not set. Using default value: 100
[ INFO] [1594798584.613176319]: Using planning request adapter 'Add Time Parameterization'
[ INFO] [1594798584.613183377]: Using planning request adapter 'Fix Workspace Bounds'
[ INFO] [1594798584.613190071]: Using planning request adapter 'Fix Start State Bounds'
[ INFO] [1594798584.613196109]: Using planning request adapter 'Fix Start State In Collision'
[ INFO] [1594798584.613202717]: Using planning request adapter 'Fix Start State Path Constraints'
[ INFO] [1594798584.888241235]: Added FollowJointTrajectory controller for 
[ INFO] [1594798585.185507661]: Added FollowJointTrajectory controller for sda10f/sda10f_r1_controller
[ INFO] [1594798585.405755680]: Stereo is NOT SUPPORTED
[ INFO] [1594798585.405799220]: OpenGl version: 4.6 (GLSL 4.6).
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
[ INFO] [1594798585.487442460]: Added FollowJointTrajectory controller for sda10f/sda10f_b1_controller
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
[ INFO] [1594798585.690816071]: Added FollowJointTrajectory controller for sda10f/sda10f_b2_controller
[ INFO] [1594798585.993892550]: Added FollowJointTrajectory controller for sda10f/sda10f_r2_controller
[ INFO] [1594798585.994011978]: Returned 5 controllers in list
[ INFO] [1594798586.011524701]: Trajectory execution is managing controllers
Loading 'move_group/ApplyPlanningSceneService'...
Loading 'move_group/ClearOctomapService'...
Loading 'move_group/MoveGroupCartesianPathService'...
Loading 'move_group/MoveGroupExecuteTrajectoryAction'...
Loading 'move_group/MoveGroupGetPlanningSceneService'...
Loading 'move_group/MoveGroupKinematicsService'...
Loading 'move_group/MoveGroupMoveAction'...
Loading 'move_group/MoveGroupPickPlaceAction'...
Loading 'move_group/MoveGroupPlanService'...
Loading 'move_group/MoveGroupQueryPlannersService'...
Loading 'move_group/MoveGroupStateValidationService'...
[ INFO] [1594798586.162801291]: 

********************************************************
* MoveGroup using: 
*     - ApplyPlanningSceneService
*     - ClearOctomapService
*     - CartesianPathService
*     - ExecuteTrajectoryAction
*     - GetPlanningSceneService
*     - KinematicsService
*     - MoveAction
*     - PickPlaceAction
*     - MotionPlanService
*     - QueryPlannersService
*     - StateValidationService
********************************************************

[ INFO] [1594798586.162957363]: MoveGroup context using planning plugin ompl_interface/OMPLPlanner
[ INFO] [1594798586.163014499]: MoveGroup context initialization complete

You can start planning now!

XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
[ INFO] [1594798588.722784452]: Loading robot model 'motoman_sda10f'...
[ WARN] [1594798588.752401585]: Kinematics solver doesn't support #attempts anymore, but only a timeout.
Please remove the parameter '/rviz_roblab4_ThinkCentre_M920s_23952_6060666620694100860/arm_left/kinematics_solver_attempts' from your configuration.
[ERROR] [1594798588.772184870]: Group 'torso' is not a chain
[ERROR] [1594798588.772206555]: Kinematics solver of type 'kdl_kinematics_plugin/KDLKinematicsPlugin' could not be initialized for group 'torso'
[ERROR] [1594798588.772239894]: Kinematics solver could not be instantiated for joint group torso.
[ERROR] [1594798588.772263891]: Group 'arms' is not a chain
[ERROR] [1594798588.772270743]: Kinematics solver of type 'kdl_kinematics_plugin/KDLKinematicsPlugin' could not be initialized for group 'arms'
[ERROR] [1594798588.772276957]: Kinematics solver could not be instantiated for joint group arms.
[ERROR] [1594798588.772293101]: Group 'sda10f' is not a chain
[ERROR] [1594798588.772301081]: Kinematics solver of type 'kdl_kinematics_plugin/KDLKinematicsPlugin' could not be initialized for group 'sda10f'
[ERROR] [1594798588.772327593]: Kinematics solver could not be instantiated for joint group sda10f.
[ INFO] [1594798588.833981118]: Starting planning scene monitor
[ INFO] [1594798588.835469148]: Listening to '/move_group/monitored_planning_scene'
QObject::connect: Cannot queue arguments of type 'QVector<int>'
(Make sure 'QVector<int>' is registered using qRegisterMetaType().)
QObject::connect: Cannot queue arguments of type 'QVector<int>'
(Make sure 'QVector<int>' is registered using qRegisterMetaType().)
[ INFO] [1594798588.927641595]: Constructing new MoveGroup connection for group 'arm_left' in namespace ''
[ INFO] [1594798590.004760703]: Ready to take commands for planning group arm_left.
[ INFO] [1594798590.004939751]: Looking around: no
[ INFO] [1594798590.005034195]: Replanning: no
[ INFO] [1594798624.925935806]: Combined planning and execution request received for MoveGroup action. Forwarding to planning and execution pipeline.
[ INFO] [1594798624.926409601]: Planning attempt 1 of at most 1
[ INFO] [1594798624.929587510]: Planner configuration 'arm_left' will use planner 'geometric::RRTConnect'. Additional configuration parameters will be set when the planner is constructed.
[ INFO] [1594798624.931823564]: RRTConnect: Starting planning with 1 states already in datastructure
[ INFO] [1594798624.932004792]: RRTConnect: Starting planning with 1 states already in datastructure
[ INFO] [1594798624.932332723]: RRTConnect: Starting planning with 1 states already in datastructure
[ INFO] [1594798624.932500256]: RRTConnect: Starting planning with 1 states already in datastructure
[ INFO] [1594798624.943263284]: RRTConnect: Created 5 states (2 start + 3 goal)
[ INFO] [1594798624.943567633]: RRTConnect: Created 5 states (2 start + 3 goal)
[ INFO] [1594798624.943911266]: RRTConnect: Created 4 states (2 start + 2 goal)
[ INFO] [1594798624.944059510]: RRTConnect: Created 5 states (2 start + 3 goal)
[ INFO] [1594798624.944677472]: ParallelPlan::solve(): Solution found by one or more threads in 0.013650 seconds
[ INFO] [1594798624.945398457]: RRTConnect: Starting planning with 1 states already in datastructure
[ INFO] [1594798624.945575409]: RRTConnect: Starting planning with 1 states already in datastructure
[ INFO] [1594798624.945741560]: RRTConnect: Starting planning with 1 states already in datastructure
[ INFO] [1594798624.945863224]: RRTConnect: Starting planning with 1 states already in datastructure
[ INFO] [1594798624.946305190]: RRTConnect: Created 4 states (2 start + 2 goal)
[ INFO] [1594798624.946531263]: RRTConnect: Created 4 states (2 start + 2 goal)
[ INFO] [1594798624.946786482]: RRTConnect: Created 4 states (2 start + 2 goal)
[ INFO] [1594798624.946910712]: RRTConnect: Created 5 states (2 start + 3 goal)
[ INFO] [1594798624.947838601]: ParallelPlan::solve(): Solution found by one or more threads in 0.002588 seconds
[ INFO] [1594798624.948318150]: RRTConnect: Starting planning with 1 states already in datastructure
[ INFO] [1594798624.948480069]: RRTConnect: Starting planning with 1 states already in datastructure
[ INFO] [1594798624.949499425]: RRTConnect: Created 4 states (2 start + 2 goal)
[ INFO] [1594798624.949628483]: RRTConnect: Created 4 states (2 start + 2 goal)
[ INFO] [1594798624.950468908]: ParallelPlan::solve(): Solution found by one or more threads in 0.002298 seconds
[ INFO] [1594798624.961917241]: SimpleSetup: Path simplification took 0.011125 seconds and changed from 3 to 2 states
[ INFO] [1594798624.964655970]: Disabling trajectory recording
```

</details>

This is the rqt_graph when lunching:

![image](https://user-images.githubusercontent.com/57666167/87420722-74ba4f00-c5de-11ea-8561-aacae6086c4c.png)
 
this is the error in the log file:

```
[rosmaster.threadpool][ERROR] 2020-07-15 10:37:51,802: Traceback (most recent call last):
  File "/opt/ros/melodic/lib/python2.7/dist-packages/rosmaster/threadpool.py", line 218, in run
    result = cmd(*args)
  File "/opt/ros/melodic/lib/python2.7/dist-packages/rosmaster/master_api.py", line 210, in publisher_update_task
    ret = xmlrpcapi(api).publisherUpdate('/master', topic, pub_uris)
  File "/usr/lib/python2.7/xmlrpclib.py", line 1243, in __call__
    return self.__send(self.__name, args)
  File "/usr/lib/python2.7/xmlrpclib.py", line 1602, in __request
    verbose=self.__verbose
  File "/usr/lib/python2.7/xmlrpclib.py", line 1283, in request
    return self.single_request(host, handler, request_body, verbose)
  File "/usr/lib/python2.7/xmlrpclib.py", line 1316, in single_request
    return self.parse_response(response)
  File "/usr/lib/python2.7/xmlrpclib.py", line 1493, in parse_response
    return u.close()
  File "/usr/lib/python2.7/xmlrpclib.py", line 800, in close
    raise Fault(**self._stack[0])
Fault: <Fault -1: 'publisherUpdate: unknown method name'>
```

if anyone have any idea what to do next it will be very helpful. thank you in advance

**Comment by gavanderhoorn** [comment]
Please do not post screenshots of terminals. It's all text, so copy-paste the text from the terminal into your post.

Edit your original post using the `Edit` option from the dropdown in the top-right.

Additionally, while removing that screenshot of your terminal, add:

 - system software version of your FS100
 - information on how you installed MotoROS (built from sources, downloaded the binaries, etc)
 - which version of MotoROS you've installed
 - whether your robot has been setup by Yaskawa Motoman or not

And include:

 - *verbatim* copies of the commands (ie: `roslaunch` and/or `rosrun`) you run
 - a screenshot of `rqt_graph` after you've started everything

Without that information, we cannot help you.

**Comment by gavanderhoorn** [comment]
Thanks for the update, but you still don't show *verbatim* copy-pastes of the console output, nor the exact commands you run.

**Comment by gavanderhoorn** [comment]
@inbarbd: please post a comment next time you edit your post, as Github does not send out notifications for edits.

**Comment by gavanderhoorn** [comment]
> when run:
> 
> ```
> moveit_planning_execution.launch robot_ip:=10.0.0.2 controller:=fs100
> ```
> 
> the robot is connected and we can read information from the controller. when move the robot manually with the pendant the robot position is updating in rviz.

Is this an exact copy-paste of the command you run?

If it is, it would appear to be missing `sim:=false`, which would mean you're running not with data from the real robot, but from a simulated one.

Try to use Wireshark to make sure there is actual network traffic between your ROS PC and the FS100.

If there is, and you're certain you've actually added `sim:=false`, then attach a short Wireshark capture here so we can investigate it.

> 2\. ADD `<param name="trajectory_execution/execution_duration_monitoring" value="false" />` to `jrajectory_exexution.launch.xml`

Could you please clarify why you've done this?

With a correctly configured and working driver and MoveIt configuration this should not be necessary (it isn't necessary for the Motoman robots I work with).

**Comment by inbarbd** [comment]
Thank you for the response.

The sim argument is config in the launch file to be equal to false.

The reason i add -- <param name="trajectory_execution/execution_duration_monitoring" value="false" /> to jrajectory_exexution.launch.xml --is due to an error in the terminal.  I try to run it on two different commuters (ROS-kineck ubuntu 16.4 and ROS-melodic Ubuntu 18.04 and get the same error that led me to fond this solution.

Attached   to this message the Wiresark file records while try to execute trajectory with moveit. 

[wireshark_records.zip](https://github.com/ros-industrial/motoman/files/4964061/wireshark_records.zip)

We also try to send trajectory message direct to the controller (from python script not with moveit) and the results are the same - the message send but the robot is not move.

Please let me know if you have  idea what should be the next step.
Thank you in advance.

**Comment by EricMarcil** [comment]
I've review the first capture:

From what I can see in the first capture.

@37.985 E-Stop is removed
@44.275 Servo turned on
@44.376 Motion Possible
@57.531 Motiion Ctrl: Motion Ready (200101)
@57.669 Joint trajectory starts
@57.669 Motion Reply: Success
…
@58.213 Joint Trajectory ends: Sequence Number 3;
@58.213 Motion Reply: Success
@58.226 Status: InMotion=true
@59.951 Status: InMotion=false

I can see that the robot is not moving throughout the sequence from the Joint Sequence.

The SDA10F type robot have 4 control groups (where group 3 and 4 need the same data) and I can see that you are only trying to move one group sending the standard simple message.  I believe that you need to use the [ROS_MSG_MOTO_JOINT_TRAJ_PT_FULL_EX = 2016] command and command all 4 joints together.

I haven't used a dual robot and don't know the detail on how to make the message switch on the ROS side.  Maybe @gavanderhoorn or someone from the community with an SDA robot can confirm and help you out.

**Comment by gavanderhoorn** [comment]
> The sim argument is config in the launch file to be equal to false.

Please tell us about this sort of thing in the future.

We cannot guess what you are doing, and changing argument default values makes diagnosing issues much harder if you don't tell us about them.

> The reason i add -- to jrajectory_exexution.launch.xml --is due to an error in the terminal. 

*Which error*?

> I try to run it on two different commuters (ROS-kineck ubuntu 16.4 and ROS-melodic Ubuntu 18.04 and get the same error that led me to fond this solution.

It is not needed if everything is working correctly.

The fact you need to add it tells me something is either not correctly configured, or interaction with the driver is not done correctly.

> Attached to this message the Wiresark file records while try to execute trajectory with moveit.
> 
> [wireshark_records.zip](https://github.com/ros-industrial/motoman/files/4964061/wireshark_records.zip)
> 
> We also try to send trajectory message direct to the controller (from python script not with moveit) and the results are the same - the message send but the robot is not move.

Could you please describe the steps you took which resulted in the 4 wireshark captures you've shared?

If you used some code, please show it here.

From your minimal descriptions, it's difficult to figure out what you are doing.

**Comment by inbarbd** [comment]
* The following error occurred without adding `<param name="trajectory_execution/execution_duration_monitoring" value="false" />`

    ```
    [ERROR] [1595747455.605097855]: Controller is taking too long to execute trajectory (the expected upper bound for the trajectory execution was 2.667912 seconds). Stopping trajectory.
    ```

* I try to run this on two different computers with different ROS configuration because I want to see if the problem is the ROS melodic. the package available this Git is until ROS kinetic. But the logs are the same on both computers.

* All the Wireshark records was taken while try to execute trajectory from the Rviz (moveit) after run in the terminal:

    ```
    roslaunch motoman_sda10f_moveit_config moveit_planning_execution.launch  robot_ip:=10.0.0.2 controller:=fs100 sim:=false
    ```

* At this moment we dont use extra code only try to move the robot from the moveit interface in Rviz.
 
If you need more information please let me know.

**Comment by gavanderhoorn** [comment]
> All the Wireshark records was taken while try to execute trajectory from the Rviz (moveit) after run in the terminal:

that's great, but please describe how you use the MoveIt RViz plugin.

So which buttons do you press, which groups do you select, how do you change the target pose, etc.

It's been a long time since I've been able to test with an SDA robot, so I can't be sure whether things are correctly configured to be able to control a single group only.

**Comment by gavanderhoorn** [comment]
This could be (somehow) related to #251.

**Comment by inbarbd** [comment]
Thank you for the answer.
We didn't yet get to the problem describe issue #251. Our problem is that we cant move non of the group control of the robot. For now we try to move arm_left or arm_right.

We also get help from Yaskawa professionals and we preform several changes in the robot controller:

- The robot version was FS3.27 with  MotoRosFS100.
- Delete MotoRosFS100 from the controller and install MotoRosFS_191​.
- Upgrade the controller version to FS3.3 with MotoRosFS_191

 
In all this configuration we try to preform the follow steps:

1. Run `roslaunch motoman_sda10f_moveit_config moveit_planning_execution.launch robot_ip:=10.0.0.2 controller:=fs100 sim:=false`
2. Run `rosservice call /robot_enable`, in different terminal.
3.servo on light is on 
success: True
message: "Motoman robot is now enabled and will accept motion commands."
3. in the Rviz the selected group is `arm_left` of `arm_right` in etch attempt. 
4. Select from the Rviz GUI New 'Goal State' to be random valid.
5. Press the Plan button.
6. Press the Execute button. 

All results was the same as before  --> The robot receive the commend but No errors show in the terminal or in the roslogs.

Only after upgrading the controller version a new error occur :

```
[ INFO] [1596026066.002472947]: RRTConnect: Starting planning with 1 states already in datastructure
[ INFO] [1596026066.002679648]: RRTConnect: Starting planning with 1 states already in datastructure
[ INFO] [1596026066.003713328]: RRTConnect: Created 5 states (2 start + 3 goal)
[ INFO] [1596026066.003975286]: RRTConnect: Created 5 states (2 start + 3 goal)
[ INFO] [1596026066.005146053]: ParallelPlan::solve(): Solution found by one or more threads in 0.002826 seconds
[ INFO] [1596026066.009451122]: SimpleSetup: Path simplification took 0.003951 seconds and changed from 3 to 2 states
[ INFO] [1596026072.361043818]: Execution request received
[ERROR] [1596026072.381973959]: 0,1
[ERROR] [1596026072.382037953]: 0,1
[ERROR] [1596026072.382072539]: 0,1
[ERROR] [1596026072.382105891]: 0,1
[ERROR] [1596026072.382133661]: 0,1
[ERROR] [1596026072.382159280]: 0,1
[ERROR] [1596026072.382192109]: 0,1
[ERROR] [1596026072.382329334]: 0,1
[ERROR] [1596026072.382367289]: 0,1
[ERROR] [1596026072.382390542]: 0,1
[ERROR] [1596026072.382415638]: 0,1
```

- The following is the telnet messages :

```
roblab4@roblab4-ThinkCentre-M920s:~$ telnet
telnet> open 10.0.0.2
Trying 10.0.0.2...
Connected to 10.0.0.2.
Escape character is '^]'.

VxWorks login: MOTOMANrobot
Password: 


-> Starting new connection to the Motion Server
Creating new task: IncMoveTask
IncMoveTask Started
Creating new task: tidAddToIncQueue (groupNo = 0)
Creating new task: tidAddToIncQueue (groupNo = 1)
Creating new task: tidAddToIncQueue (groupNo = 2)
Creating new task: tidAddToIncQueue (groupNo = 3)
Creating new task: tidMotionConnections (connectionIndex = 0)
Starting new connection to the State Server
Starting State Server Send State task
Controller number of group = 4
In StartTrajMode
Robot job is ready for ROS commands.
```

No more errors or messages on the telnet connection.

This is the wireshark recorded for 3 identical attempts:
[motoman_new_vertion_ubuntu.zip](https://github.com/ros-industrial/motoman/files/4994908/motoman_new_vertion_ubuntu.zip)

 
The Robot still Not move. 

Do you have any idea what is the next steps in order to control the robot throw the Ethernet connection?
Thank you in advance,
inbar

**Comment by gavanderhoorn** [comment]
> We also get help from Yaskawa professionals

That's great, but it also makes me hesitant to continue diagnosing here.

Trying to figure out a problem like this from two sides without those sides communicating or keeping each other up-to-date is not a good way to approach this.

@EricMarcil @ted-miller: are any of you involved here?

**Comment by gavanderhoorn** [comment]
> ```
> [ERROR] [1596026072.381973959]: 0,1
> ```

I don't recognise this `ERROR` log line.

Please configure your rosconsole configuration to include the node name, so we can see which node prints this error.

Add this line to your `.bashrc`:

```
export ROSCONSOLE_FORMAT='[${severity}] [${time}] [${node}] [${file}:${line}] [${logger}]: ${message}'
```

and try running things again. The log output should now include the name of the node.

Please copy-paste *everything* from the console into a comment here.

**Comment by ted-miller** [comment]
@gavanderhoorn, I have instructed @inbarbd to use my Windows emulator application to test the robot and MotoROS application without using ROS-I. He is experiencing the same issue (no motion) using the emulator app. I can't find any obvious issues in the Wireshark captures.

Given that the emulator application is experiencing the same symptoms, I'm fairly confident that there is a problem with the configuration of the FS100. (Note: There might also be a problem with the ROS setup... but one issue at a time.) This controller has been used for other purposes in the past. It is likely that there is some safety or interlock signal that is left over from a previous application.

I have requested that they attempt to program a standard job on the teach pendant to see if motion is being blocked by something else in the sysytem. I have also asked the local Yaskawa rep to look into sending a technician to re-initialize the controller.

**Comment by gavanderhoorn** [comment]
Thanks for the update @ted-miller. In the future: please post about any interaction like this with users which are not the result of something discussed here, for obvious reasons. Private emails are of course fine -- and in some cases unavoidable -- but it's rather unfortunate if it's not needed.

Given what you write, I'm going to put my efforts on hold until we get clarity about the status of the controller @inbarbd is trying to use.

**Comment by ted-miller** [comment]
Today, it was discovered that the robot job (INIT_ROS) is setup incorrectly. Their robot job is a non-group job. The motion API will only work if the commanded control groups are sitting on a WAIT command in Inform. 

Tomorrow they are going to replace the INIT_ROS job and try again. 
![Annotation 2020-08-03 155836](https://user-images.githubusercontent.com/4623811/89221987-37742c00-d5a2-11ea-9173-d749e6bb556c.png)

**Comment by ted-miller** [comment]
After correcting the INIT_ROS job, the robot is behaving as expected.

**Comment by gavanderhoorn** [comment]
For future cases: do we know how the `INIT_ROS` job ended up groupless?

Was this a copy-pasta error, a corrupted file or something else?

**Comment by ted-miller** [comment]
There was an issue with the configuration of the group combinations, so the file from Github wouldn't load. So, the job was manually typed on the pendant as a non-group job.

**Comment by gavanderhoorn** [comment]
Ah, ok.

So the root cause was a problem with the group configuration on the controller then.

Thanks for updating the issue.
